libraries/AC_Avoidance/AP_OABendyRuler.h

```diff
@@ -10,25 +10,29 @@
  */
 class AP_OABendyRuler {
 public:
-
-    AP_OABendyRuler() {}
+    AP_OABendyRuler();
 
     /* Do not allow copies */
     AP_OABendyRuler(const AP_OABendyRuler &other) = delete;
     AP_OABendyRuler &operator=(const AP_OABendyRuler&) = delete;
 
     // send configuration info stored in front end parameters
-    void set_config(float lookahead, float margin_max) { _lookahead = MAX(lookahead, 1.0f); _margin_max = MAX(margin_max, 0.0f); }
+    void set_config(float margin_max) { _margin_max = MAX(margin_max, 0.0f); }
 
     // run background task to find best path and update avoidance_results
     // returns true and populates origin_new and destination_new if OA is required.  returns false if OA is not required
     bool update(const Location& current_loc, const Location& destination, const Vector2f &ground_speed_vec, Location &origin_new, Location &destination_new);
 
+    static const struct AP_Param::GroupInfo var_info[];
+
 private:
 
     // calculate minimum distance between a path and any obstacle
     float calc_avoidance_margin(const Location &start, const Location &end);
 
+    // determine if BendyRuler should accept the new bearing or try and resist it. Returns true if bearing is not changed  
+    bool avoid_bearing_change(const Location &destination, const Location &current_loc, const bool &active, const float &bearing_test, const float &lookahead_step1_dist, const float &margin, float &final_bearing, float &final_margin);
```

Comment:
as mentioned let's rename this to "ignore_bearing_change" or "resist_bearing_change" or something like that.